src/wtforms/validators.py

```diff
@@ -521,7 +521,7 @@ def __call__(self, form, field):
         if message is None:
             message = field.gettext("Invalid UUID.")
         try:
-            uuid.UUID(field.data)
+            uuid.UUID(str(field.data))
         except ValueError:
             raise ValidationError(message)
```

Comment:
It might be best to check if the type is uuid.UUID instead of blindly casting to str

```suggestion
        if isinstance(field.data, uuid.UUID):
            return

        try:
            uuid.UUID(field.data)
        except ValueError:
            raise ValidationError(message)
```